include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,1013 @@
+            container::HostBuffer<float_64, DIM1> reducedSumMomPos2( subGrid.getLocalDomain().size.y() );
+            container::HostBuffer<float_64, DIM1> reducedCount_e( subGrid.getLocalDomain().size.y() );
+            reducedSumMom2.assign( 0.0 );
+            reducedSumPos2.assign( 0.0 );
+            reducedSumMomPos2.assign( 0.0 );
+            reducedCount_e.assign( 0.0 );
+
+            // add gSum values from all GPUs using MPI
+            planeReduce->template operator()( /* parameters: dest, source */
+                             reducedSumMom2,
+                             gSumMom2->getHostBuffer( ).cartBuffer(),
+                             /* the functors return value will be written to dst */
+                             pmacc::algorithm::functor::Add() );
+            planeReduce->template operator()( /* parameters: dest, source */
+                             reducedSumPos2,
+                             gSumPos2->getHostBuffer( ).cartBuffer(),
+                             /* the functors return value will be written to dst */
+                             pmacc::algorithm::functor::Add() );
+            planeReduce->template operator()( /* parameters: dest, source */
+                             reducedSumMomPos2,
+                             gSumMomPos2->getHostBuffer( ).cartBuffer(),
+                             /* the functors return value will be written to dst */
+                             pmacc::algorithm::functor::Add() );
+            planeReduce->template operator()( /* parameters: dest, source */
+                             reducedCount_e,
+                             gCount_e->getHostBuffer( ).cartBuffer(),
+                             /* the functors return value will be written to dst */
+                             pmacc::algorithm::functor::Add() );
+
+            /** all non-reduce-root processes are done now */
+            if( !this->isPlaneReduceRoot )
+                return;
+
+            // gather to file writer
+            container::HostBuffer< float_64, DIM1 > globalSumMom2( subGrid.getGlobalDomain().size.y() );
+            container::HostBuffer< float_64, DIM1 > globalSumPos2( subGrid.getGlobalDomain().size.y() );
+            container::HostBuffer< float_64, DIM1 > globalSumMomPos2( subGrid.getGlobalDomain().size.y() );
+            container::HostBuffer< float_64, DIM1 > globalCount_e( subGrid.getGlobalDomain().size.y() );
+
+            // gather y offsets, so we can store our gathered data in the right order
+            int gatherSize = -1;
+            int gatherRank = -1;
+            MPI_CHECK(MPI_Comm_rank( commGather, &gatherRank ));
+            MPI_CHECK(MPI_Comm_size( commGather, &gatherSize ));
+            std::vector< uint64_t > y_offsets( gatherSize );
+            std::vector< int > y_sizes( gatherSize );
+            std::vector< int > gatherRanks( gatherSize );
+            long int const y_off = subGrid.getGlobalDomain().offset.y();
```

Comment:
huh, shouldn't this be `getLocalDomain().size.y()`?

Reply:
@ax3l and @kossag14 was this correct (and tested)?